libsolidity/parsing/Parser.cpp

```diff
@@ -131,6 +131,9 @@ ASTPointer<SourceUnit> Parser::parse(CharStream& _charStream)
 			case Token::Function:
 				nodes.push_back(parseFunctionDefinition(true));
 				break;
+			case Token::Event:
+				nodes.push_back(parseEventDefinition());
+				break;
```

Comment:
This is suspiciously simple, but then again, why wouldn't it be :)

Reply:
Yeah, I thought the same thing :) But in the end I could not come up with a test case that would break it so it seems that it's all that's needed. It makes sense, because it was a bit of an artificial limitation - it does not allow you to do new things in terms of code generation. I was mostly worried about naming conflicts in Natspec and ABI bit, but the hack from #13086 seems to have solved all issues there.